python/pyspark/sql/tests.py

```diff
@@ -3654,6 +3654,107 @@ def test_jvm_default_session_already_set(self):
             spark.stop()
 
 
+class SparkSessionTests2(ReusedSQLTestCase):
```

Comment:
Do we need to extend `ReusedSQLTestCase`? Looks we can just `unittest.TestCase`.

Reply:
@HyukjinKwon there's no strong need for it, however it does mean that the first `getOrCreate` will already have a session it can use, but given that we set up and tear down the session this may be less than ideal.